packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -0,0 +1,93 @@
+import { Fn } from './cfn-fn';
+import { Aws } from './cfn-pseudo';
+
+/*
+ * This file contains functions for translating from a pure CFN value
+ * (like a JS object { "Ref": "Bucket" })
+ * to a form CDK understands
+ * (like Fn.ref('Bucket')).
+ */
+
+export function cfnValueToCdkValue(cfnValue: any): any {
+  if (cfnValue === undefined || cfnValue === null) {
+    return undefined;
+  }
+  if (typeof cfnValue === 'string' || typeof cfnValue === 'number') {
```

Comment:
No love for dem `bool`s? 

Honesty I would not put this clause here and just stick an unconditional return for anything that's not "special" at the end.

Reply:
Yep. Done as part of https://github.com/aws/aws-cdk/pull/7304#discussion_r420418178